src/tests/ftest/util/agent_utils.py

```diff
@@ -224,6 +229,53 @@ def start(self):
 
         super().start()
 
+    def dump_attachinfo(self):
+        """Run dump-attachinfo on the daos_agent."""
+        self.manager.job.set_sub_command("dump-attachinfo")
+        self.manager.job.sudo = True
+        self.attachinfo = run_pcmd(self.hosts,
+                                   str(self.manager.job))[0]["stdout"]
+        self.log.info("Agent attachinfo: %s", self.attachinfo)
+
+    def get_attachinfo_file(self):
+        """Run dump-attachinfo on the daos_agent."""
+
+        self.manager.job.set_sub_command("dump-attachinfo")
+        self.manager.job.sudo = True
+        self.attachinfo = run_pcmd(self.hosts,
+                                   str(self.manager.job))[0]["stdout"]
+        self.log.info("Agent attachinfo: %s", self.attachinfo)
+
+        server_name = "daos_server"
+
+        self.dump_attachinfo()
+        self.log.info("Agent attachinfo: %s",
+                      self.attachinfo)
+
+        a = self.attachinfo
+
+        # Filter log messages from attachinfo content
+        L = [x for x in a if re.match(r"^(name\s|size\s|all|\d+\s)", x)]
+        attach_info_contents = "\n".join(L)
+        attach_info_filename = "{}.attach_info_tmp".format(server_name)
+
+        if len(L) < 4:
+            self.log.info("Malformed attachinfo file: %s",
+                          attach_info_contents)
+            return None
+
+        # Write an attach_info_tmp file in this directory for cart_ctl to use
+        tmpdir = tempfile.gettempdir()
+        attachinfo_file_path = os.path.join(tmpdir, attach_info_filename)
+
```

Comment:
We really should use the avocado provided directory that will also enable us to retain a copy for inspection after the test completes:
```suggestion
        attachinfo_file_path = os.path.join(self.outputdir, attach_info_filename)

```